tests/components/sql/test_config_flow.py

```diff
@@ -594,3 +609,93 @@ async def test_full_flow_not_recorder_db(
         "column": "value",
         "unit_of_measurement": "MB",
     }
+
+
+async def test_device_state_class(recorder_mock: Recorder, hass: HomeAssistant) -> None:
+    """Test we get the form."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={},
+        options={
+            "name": "Get Value",
+            "query": "SELECT 5 as value",
+            "column": "value",
+            "unit_of_measurement": "MiB",
+        },
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "init"
+
+    with patch(
+        "homeassistant.components.sql.async_setup_entry",
+        return_value=True,
+    ):
+        result2 = await hass.config_entries.options.async_configure(
+            result["flow_id"],
+            user_input={
+                "query": "SELECT 5 as value",
+                "column": "value",
+                "unit_of_measurement": "MiB",
+                "device_class": SensorDeviceClass.DATA_SIZE,
+                "state_class": SensorStateClass.TOTAL,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["data"] == {
+        "name": "Get Value",
+        "query": "SELECT 5 as value",
+        "column": "value",
+        "unit_of_measurement": "MiB",
+        "device_class": SensorDeviceClass.DATA_SIZE,
+        "state_class": SensorStateClass.TOTAL,
+    }
+
+    await init_integration(hass, result2["data"], entry_id=entry.entry_id)
```

Comment:
Please make this it's own test in `test_sensor.py`
Thanks

Reply:
Added in https://github.com/home-assistant/core/pull/95020/commits/36188946c223943cd2b6b1da52cc75e3b8afaf43, I had included it in `test_config_flow.py` because in `test_sensor.py` I can't validate if the configuration option manipulations are reaching the final result in the sensor (unless I reproduce the same test `test_device_state_class` in `test_sensor.py`).